src/main/haskell/matching-logic/test/Kore/MatchingLogic/ProofSystem/OnePlusOne.hs

```diff
@@ -255,7 +255,7 @@ convertRule convertHyp r = case r of
     Ax_propositional1 f1 f2 -> Propositional1 (convertFormula f1) (convertFormula f2)
     Ax_propositional2 f1 f2 f3 -> Propositional2 (convertFormula f1) (convertFormula f2) (convertFormula f3)
     Ax_propositional3 f1 f2 -> Propositional3 (convertFormula f1) (convertFormula f2)
-    Ax_varSubst x y p1 p2 -> VariableSubstitution (SubstitutedVariable (nat2Int x))
+    Ax_varSubst x y p1 _ -> VariableSubstitution (SubstitutedVariable (nat2Int x))
```

Comment:
It's a bit strange that p2 is not needed here, I think that it's the only rule where one of the pieces is ignored. On the other hand, I think that the ML variable substitution rule had only one pattern, so I guess it might be fine.

@bmmoore What does p2 mean here?

Reply:
p2 here is the result of the substitution, but the proof checking for VariableSubstitution always computes the expected result with varSubst. The p2 parameter exists here because the Coq definition defined substitution as a relation, initially to reflect the matching logic paper allowing capture-avoiding substitution and also to more easily experiment with variants on side conditions prohibiting capture.

In the end the conditions we already put on the VariableSubstitution rule did not interfere too badly, and this translation should preserve the conclusion of the proof rule (the overall consistency of the translated proof is checked when the proof object is assembled).